libs/common-utils/src/units.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Lightweight, dependency-free formatter that converts atom amounts to human-readable units.
+ * - Accepts string | number | bigint inputs
+ * - Handles negatives and trims trailing zeros in the fractional part
+ * - Returns an empty string for undefined/null inputs
+ */
+export function formatUnitsSafe(value: string | number | bigint | undefined | null, decimals?: number): string {
```

Comment:
Isn't `formatAmountWithPrecision()` the same?

Reply:
> Not quite the same, formatAmountWithPrecision expects a FractionLike, applies locale-aware rounding and suffix logic, and returns '0' when the amount is zero. Here we’re dealing with raw integer-like atom values (string/number/bigint) and need strict validation plus a simple decimals split (including negatives and empty fallback). That’s why formatUnitsSafe is implemented separately.